### PR TITLE
add setting MinResources to pg for normal pod

### PR DESCRIPTION
**Pull request by huone1**
Signed-off-by: huone1 <huwanxing@huawei.com>
fix issue #1542 
there are two problem:
1. minresource in podgroup for deployment is empty
2. the podgroup init status is "" and volcano scheduler doesn't adjust whether the status is "", So it isgoing to cause a lot of problems. such as queue capability failure

### CODE REVIEW

File: pkg/scheduler/actions/allocate/allocate.go [hunk 1]
```diff
@@ -58,7 +58,9 @@ func (alloc *Action) Execute(ssn *framework.Session) {
 	jobsMap := map[api.NamespaceName]map[api.QueueID]*util.PriorityQueue{}
 
 	for _, job := range ssn.Jobs {
-		if job.PodGroup.Status.Phase == scheduling.PodGroupPending {
+		if job.PodGroup.Status.Phase == scheduling.PodGroupPending || job.PodGroup.Status.Phase == "" {
```
Comment (shinytang6) [inline pkg/scheduler/actions/allocate/allocate.go:61]: l believe there are more code having this issue, pls help check it out.
and it's better to abstract a func to do that

**Review by shinytang6 — CHANGES_REQUESTED** [review]

File: pkg/scheduler/actions/allocate/allocate.go (hunk 1 above)
Comment (huone1) [reply pkg/scheduler/actions/allocate/allocate.go]: OK, define the function "jobinfo.IsPending() bool" to check it.

File: pkg/scheduler/actions/enqueue/enqueue.go [hunk 2]
```diff
@@ -66,6 +66,10 @@ func (enqueue *Action) Execute(ssn *framework.Session) {
 			queues.Push(queue)
 		}
 
+		if job.PodGroup.Status.Phase == "" {
+			continue
+		}
+
 		if job.PodGroup.Status.Phase == scheduling.PodGroupPending {
```
Comment (william-wang) [inline pkg/scheduler/actions/enqueue/enqueue.go:73]: Use job.IsPending() to replace 'job.PodGroup.Status.Phase == scheduling.PodGroupPending' here.

Comment (huone1) [reply pkg/scheduler/actions/enqueue/enqueue.go:73]: don't do it, because the pending job  should be  pushed  to jobsMap.

**Review by shinytang6 — APPROVED** [review]
/lgtm

File: pkg/scheduler/actions/enqueue/enqueue.go [hunk 3]
```diff
@@ -65,6 +65,9 @@ func (enqueue *Action) Execute(ssn *framework.Session) {
 			queueMap[queue.UID] = queue
 			queues.Push(queue)
 		}
+		if job.PodGroup.Status.Phase == "" {
```
Comment (Thor-wl) [inline pkg/scheduler/actions/enqueue/enqueue.go:68]: Suggest to add some comments here in order not to make confusion for developers later.

File: pkg/scheduler/api/job_info.go [hunk 4]
```diff
@@ -589,3 +589,16 @@ func (ji *JobInfo) Ready() bool {
 
 	return occupied >= ji.MinAvailable
 }
+
+// IsPending returns whether job is in pending status
+func (ji *JobInfo) IsPending() bool {
```
Comment (Thor-wl) [inline pkg/scheduler/api/job_info.go:594]: Maybe it can be merged with the `if` below.

File: pkg/scheduler/plugins/proportion/proportion.go [hunk 5]
```diff
@@ -263,9 +262,14 @@ func (pp *proportionPlugin) OnSessionOpen(ssn *framework.Session) {
 		}
 
 		if job.PodGroup.Spec.MinResources == nil {
+			klog.V(4).Infof("job %s MinResources is null.", job.Name)
 			return util.Permit
 		}
 		minReq := job.GetMinResources()
+
+		klog.V(4).Infof("job %s min resource<%v>", job.Name, minReq)
```
Comment (Thor-wl) [inline pkg/scheduler/plugins/proportion/proportion.go:270]: These logs can be removed which are just for test

Comment (huone1) [reply pkg/scheduler/plugins/proportion/proportion.go:270]: not only for test，but also for debug when the plugin behavior doesn't meet expectations. And the log level is changed to 5

File: pkg/controllers/podgroup/pg_controller_handler.go [hunk 6]
```diff
@@ -141,3 +142,42 @@ func newPGOwnerReferences(pod *v1.Pod) []metav1.OwnerReference {
...
+			list[name] = quantity.DeepCopy()
+		} else {
+			value.Add(quantity)
+			list[name] = value
+		}
+	}
+}
+
+// calcPGMinResources calculate podgroup minimum resource
+func calcPGMinResources(pod *v1.Pod) *v1.ResourceList {
```
[the remark this answers is not in the document]
Comment (huone1) [reply pkg/controllers/podgroup/pg_controller_handler.go:175]: the request resource of BestEffort  pod is empty  and minresource is empty, so there is no limit about the resource.

File: pkg/scheduler/actions/enqueue/enqueue.go (hunk 3 above)
Comment (huone1) [reply pkg/scheduler/actions/enqueue/enqueue.go]: it  is  modified to `if job.IsPending()` with the if below

File: pkg/scheduler/api/job_info.go (hunk 4 above)
Comment (huone1) [reply pkg/scheduler/api/job_info.go:594]: merged it

**Comment by jasonliu747** [comment]
/hold
@huone1 let us know when you feel ready~

**Review by shinytang6 — CHANGES_REQUESTED** [review]

File: installer/helm/chart/volcano/config/volcano-scheduler-ci.conf [hunk 7]
```diff
@@ -6,9 +6,9 @@ tiers:
   - name: conformance
   - name: sla
 - plugins:
+  - name: overcommit
```
[the remark this answers is not in the document]
Comment (huone1) [reply installer/helm/chart/volcano/config/volcano-scheduler-ci.conf:9]: plugin overcommit just  has the function jobEnqueuedFns , it shouldn't have any conflict with reclaim or preempt

**Review by Thor-wl — APPROVED** [review]

**Review by jasonliu747 — COMMENTED** [review]
/lgtm

**Pull request merged**